openquake/commonlib/readinput.py

```diff
@@ -670,12 +672,20 @@ def get_exposure_lazy(fname, ok_cost_types):
         area = conversions.area
     except NameError:
         area = LiteralNode('area', dict(type=''))
+
+    # read the cost types and make some check
     cost_types = [(ct['name'], ct['type'], ct['unit'])
                   for ct in conversions.costTypes
                   if ct['name'] in ok_cost_types]
     if 'occupants' in ok_cost_types:
         cost_types.append(('occupants', 'per_area', 'people'))
     cost_types.sort(key=operator.itemgetter(0))
+    for row in cost_types:
+        valid.simple_id(row[0])  # the name
+        for col in row:
+            if len(col) > COST_TYPE_SIZE:
```

Comment:
You are right that the test should be done earlier. Let me see if I can move this up in the validation chain (right now there is no check at all).